### DISCUSSION TITLE
Can we use first or second lagrange variable in THM?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#25529] [Q&A General]
@joshuahansel Hi Josh,
As the title, can we set the solution variable to first or second lagrange? If so, is there anything that requires additional development?

Ethan

**Comment by joshuahansel** [comment]
If you're asking about the flow channel components, you're stuck with `CONST MONOMIAL` because the spatial discretization is actually a finite volume scheme. To use a different FE type, you'd need a corresponding stabilization. We used to have `FIRST LAGRANGE` in the beginning with an artificial viscosity, but it was a bit problem-dependent to get it right.

**Reply by Ethan-xj** [reply]
Yes flow channel components. I want to find out if higher-order discretization lead to more precise calculations. Btw, where can I change the order of the variable?

**Reply by GiudGiud** [reply]
The variables are created by the component. You could change the order in the code for sure, but again it wont necessarily converge. 

I think refining the mesh is a more supported way of examining the spatial discretization errors in your calculation

**Reply by joshuahansel** [reply]
If you tried to change the variable order/family with our current methods, I just don't know what would happen. Immediate error maybe, or maybe just silent garbage.

But @GiudGiud is right, refining the mesh is your best bet. There are slope reconstruction options as well. There's a parameter `rdg_slope_reconstruction = FULL` you can use in the flow channel to get 2nd order, which will reduce mesh discretization much quicker.

**Reply by lindsayad** [reply]
As @joshuahansel said the implementation is FV/rDG, *not* DG. So you cannot adjust the convergence order by changing the polynomial order of the variable basis. You would have to be able to change the order of the reconstruction; one of the ways is to do what @joshuahansel mentioned with the full reconstruction. But fundamentally you will be limited at the upper end to a second order accurate reconstruction (with respect to an L2 norm)
